cmd/notation/internal/truststore/truststore.go

```diff
@@ -37,17 +37,23 @@ func AddCert(path, storeType, namedStore string, display bool) error {
 		return errors.New("named store name needs to follow [a-zA-Z0-9_.-]+ format")
 	}
 
-	// check if the target path is a cert (support PEM and DER formats)
-	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+	// check if the target path is a x509 certificate
+	// (support PEM and DER formats)
+	certs, err := corex509.ReadCertificateFile(certPath)
+	if err != nil {
 		return err
 	}
+	if len(certs) == 0 {
+		return errors.New("no valid certificate found in the empty file")
```

Comment:
It might not be an empty file. It can be a non-empty file with no certs.

Reply:
Oh yeah, that's true.